sys/linux/dev_sequencer.txt

```diff
@@ -0,0 +1,203 @@
+ioctl$SNDCTL_FM_LOAD_INSTR(fd fd_seq, cmd const[SNDCTL_FM_LOAD_INSTR])
+
+synth_info {
+	name		array[int8, 30]
+	device		int32
+	synth_type	flags[synth_type_flags, int8]
+	synth_subtype	flags[synth_subtype_flags, int32]
+	perc_mode	int32
+	nr_voices	int32
+	nr_drums	int32
+	instr_bank_size	int32
+	capabilities	flags[capabilities_flags, int64]
+	dummies		array[int32, 19]
+}
+
+midi_info {
+	name		array[int8, 30]
+	device		int32
+	capabilities	int32
+	dev_type	int32
+	dummies		array[int32, 18]
+}
+
+evrec_write [
+	s	evrec_short
+	n	evrec_note
+	l	evrec_long
+	v	evrec_voice
+	t	evrec_timer
+	e	evrec_extended
+	x	evrec_sysex
+	echo	int32
+	raw	evrec_raw
+	generic	evrec_generic
+] [varlen]
+
+evrec [
+	s	evrec_short
+	n	evrec_note
+	l	evrec_long
+	v	evrec_voice
+	t	evrec_timer
+	e	evrec_extended
+	x	evrec_sysex
+	echo	int32
+	raw	evrec_raw
+	generic	evrec_generic
+]
+
+evrec_short {
+	code	const[SEQ_MIDIPUTC, int8]
+	parm1	evrec_short_parm1
+	dev	int8[0:SNDRV_SEQ_OSS_MAX_MIDI_DEVS]
+	parm2	int8
+}
+
+evrec_short_parm1 [
+	SEQ_MIDIPUTC	flags[evrec_short_parm1_flags, int8]
+	generic		int8
+]
+
+evrec_note {
+	code	flags[evrec_note_code, int8]
+	chn	int8
+	note	evrec_note_note
+	vel	int8
+}
+
+evrec_note_note [
+	SEQ_NOTEON	flags[note_on_event_flags, int8]
+	generic		int8
+]
+
+evrec_timer {
+	code	flags[evrec_timer_code, int8]
+	cmd	flags[evrec_timer_cmd, int8]
+	dummy1	int8
+	dummy2	int8
+	time	int32
+}
+
+evrec_extended {
+	code	const[SEQ_EXTENDED, int8]
+	cmd	flags[evrec_extended_cmd, int8]
+	dev	int8[0:15]
+	chn	int8
+	p1	evrec_extended_p1
+	p2	int8
+	p3	int8
+	p4	int8
+}
+
+evrec_extended_p1 [
+	SEQ_CONTROLLER	flags[evrec_extended_p1_flags, int8]
+	SEQ_NOTEON	flags[note_on_event_flags, int8]
+	generic		int8
+]
+
+evrec_long {
+	code	const[EV_CHN_COMMON, int8]
+	dev	int8[0:15]
+	cmd	flags[evrec_long_cmd, int8]
+	chn	int8[0:31]
+	p1	int8
+	p2	int8
+	val	int16
+}
+
+evrec_voice {
+	code	const[EV_CHN_VOICE, int8]
+	dev	int8[0:15]
+	cmd	flags[evrec_voice_cmd, int8]
+	chn	int8[0:31]
+	note	evrec_voice_note
+	parm	int8
+	dummy	int16
+}
+
+evrec_voice_note [
+	MIDI_NOTEON	flags[note_on_event_flags, int8]
+	generic		int8
+]
+
+evrec_sysex {
+	code	const[EV_SYSEX, int8]
+	dev	int8[0:15]
+	buf	array[int8, 6]
+}
+
+evrec_raw {
+	code	flags[evrec_raw_code, int8]
+	dev	int8[0:15]
+	arr	array[int8, 6]
+}
+
+evrec_generic {
+	code	flags[evrec_generic_flags, int8]
+}
+
+synth_type_flags = SYNTH_TYPE_FM, SYNTH_TYPE_SAMPLE, SYNTH_TYPE_MIDI
+synth_subtype_flags = FM_TYPE_ADLIB, FM_TYPE_OPL3, MIDI_TYPE_MPU401
+capabilities_flags = SYNTH_CAP_PERCMODE, SYNTH_CAP_OPL3, SYNTH_CAP_INPUT
+
+evrec_extended_cmd = SEQ_NOTEOFF, SEQ_NOTEON, SEQ_PGMCHANGE, SEQ_AFTERTOUCH, SEQ_BALANCE, SEQ_CONTROLLER, SEQ_VOLMODE
+evrec_extended_p1_flags = CTRL_PITCH_BENDER, CTRL_PITCH_BENDER_RANGE
+evrec_voice_cmd = MIDI_NOTEON, MIDI_NOTEOFF, MIDI_KEY_PRESSURE
+evrec_timer_code = EV_TIMING, SEQ_SYNCTIMER
+evrec_timer_cmd = TMR_ECHO, TMR_STOP, TMR_CONTINUE, TMR_TEMPO
```

Comment:
We may need more: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/soundcard.h#L1024

Usage: https://elixir.bootlin.com/linux/latest/source/sound/core/seq/oss/seq_oss_timer.c#L79